templates/pkg/crd_sdk.go.tpl

```diff
@@ -369,3 +370,66 @@ func (rm *resourceManager) setStatusDefaults (
 		ko.Status.Conditions = []*ackv1alpha1.Condition{}
 	}
 }
+
+// updateConditions returns updated resource, true; if conditions were updated
+// else it returns nil, false
+func (rm *resourceManager) updateConditions (
+	r *resource,
+	err error,
+) (*resource, bool) {
+	ko := r.ko.DeepCopy()
+	rm.setStatusDefaults(ko)
+
+	// Terminal condition
+	var terminalCondition *ackv1alpha1.Condition = nil
+	for _, condition := range ko.Status.Conditions {
+		if condition.Type == ackv1alpha1.ConditionTypeTerminal {
+			terminalCondition = condition
+			break
+		}
+	}
+	if terminalCondition == nil {
+		terminalCondition = &ackv1alpha1.Condition{
+			Type:   ackv1alpha1.ConditionTypeTerminal,
+		}
+		ko.Status.Conditions = append(ko.Status.Conditions, terminalCondition)
+	}
+	if terminalErr, ok := rm.terminalAWSError(err); ok {
```

Comment:
Given my advice below about making the `terminalAWSError()` function only return a `bool`, you would change the above to this instead, which I think is a bit easier to read:

```go
if rm.isTerminalError(err) {
    terminalCondition.Status = corev1.ConditionTrue
    awsErr, _ := ackerr.AWSError(err)
    errorMessage := awsErr.Message()
    terminalCondition.Message = &errorMessage
}
```

Reply:
Updated code template.